pandas/tests/frame/test_asof.py

```diff
@@ -106,3 +106,30 @@ def test_all_nans(self):
         result = DataFrame(np.nan, index=[1, 2], columns=['A', 'B']).asof(3)
         expected = Series(np.nan, index=['A', 'B'], name=3)
         tm.assert_series_equal(result, expected)
+
+    def test_time_zone_aware_index(self):
+        # GH21194
+        timestamp1 = Timestamp('2018-01-01 21:00:05.001+00:00')
+        timestamp2 = Timestamp('2018-01-01 22:35:10.550+00:00')
+        timestamp_internal = timestamp2 + ((timestamp2 - timestamp1) / 2)
+
+        # Testing if DataFrame index preserve timezone (UTC)
+        df = DataFrame(data=[1, 2], index=[timestamp1, timestamp2])
+        result = df.asof(timestamp_internal)
+        expected = Series(2.0, index=RangeIndex(start=0, stop=1, step=1),
```

Comment:
`RangeIndex` isn't explicit needed to construct the Series I think. It's created by default.